Census API to Socrata/PythonCensusApiCityReader.py

```diff
@@ -30,31 +32,33 @@ def close(self):
 				feature_name = ""
 				feature_value = ""
 				indicator_url = ""
-				indicator_url = self.url + year + '/acs1?get=NAME,' + self.indicators + '&for=place:' + fips_city + '&in=state:' + fips_state + '&key=' + self.apikey
-				# self.log.logMessageString("indicator_url="+indicator_url, fmeobjects.FME_WARN)
+				indicator_url = self.url + year + '/acs' + self.acsyear + '?get=NAME,' + self.indicators + '&for=place:' + fips_city + '&in=state:' + fips_state + '&key=' + self.apikey
+				self.log.logMessageString(indicator_url)
 				req = urllib2.Request(indicator_url)
 				opener = urllib2.build_opener()
-				# make request to Census
-				f = opener.open(req)
-				# load the JSON response
-				response = json.loads(f.read())
-				# for debugging, uncomment logging and Warning message will appear in FME console when run
-				# self.log.logMessageString("response="+str(response), fmeobjects.FME_WARN)
-				# if response consists of more than 1 record (first record consists of columns names), then return the row data to FME
-				if len(response)>1:
-					# loop through the rows of the response
-					for n in range(len(response)):
-						# create the FME feature object to store attributes to output
-						feature = fmeobjects.FMEFeature()
-						row = response[n]
-						# loop through the columns of each row so we can output the values to FME - Note, all attributes need to be explicitly set in PythonCreator
-						for i in range(len(row)):
-							feature_name = "col" + str(i)
-							feature_value = row[i]
-							if feature_value == None:
-								feature_value = ""
-							feature.setAttribute(feature_name, feature_value)
-						# output year
-						feature.setAttribute("Year", year)
-						# output each row's set of features to FME
-						self.pyoutput(feature)
+				# make request to Census API
+				try:
+					f = opener.open(req)
+					# load the JSON response
+					response = json.loads(f.read())
+					# if response consists of more than 1 record (first record consists of columns names), then return the row data to FME
+					if len(response)>1:
+						# loop through the rows of the response
+						for n in range(len(response)):
+							# create the FME feature object to store attributes to output
+							feature = fmeobjects.FMEFeature()
+							row = response[n]
+							# loop through the columns of each row so we can output the values to FME - Note, all attributes need to be explicitly set in PythonCreator
+							for i in range(len(row)):
+								feature_name = "col" + str(i)
+								feature_value = row[i]
+								if feature_value == None:
+									feature_value = ""
+								feature.setAttribute(feature_name, feature_value)
+							# output year
+							feature.setAttribute("Year", year)
+							# output each row's set of features to FME
+							self.pyoutput(feature)
+				except URLError, e:
```

Comment:
Comma syntax is fine since I'm assuming we don't know which Python environment this script will be run on. However if we are certain Python 2.6+ is involved, we should change this to `except URLError as e` to avoid potential confusion.